pkg/fieldpath/paved.go

```diff
@@ -483,6 +484,17 @@ func (p *Paved) SetValue(path string, value any) error {
 	return p.setValue(segments, value)
 }
 
+func (p *Paved) validateSegments(s Segments) error {
+	if p.maxFieldPathIndexEnabled() {
+		for _, segment := range s {
+			if segment.Type == SegmentIndex && segment.Index > p.maxFieldPathIndex {
+				return errors.Errorf("index %v is greater than max allowed index %d", segment.Index, p.maxFieldPathIndex)
+			}
+		}
+	}
+	return nil
```

Comment:
nit for readability: 

```suggestion
        if !p.maxFieldPathIndexEnabled() {
            return nil
        }
        for _, segment := range s {
	        if segment.Type == SegmentIndex && segment.Index > p.maxFieldPathIndex {
		        return errors.Errorf("index %v is greater than max allowed index %d", segment.Index, p.maxFieldPathIndex)
	        }
        }
	return nil
```

Reply:
generally agree with that, but in this case I wanted to keep it nested so that in the future additional checks could be added without changing the existing code

Reply:
still, changed accordingly